sound/soc/sof/sof-client-dma-trace.c

```diff
@@ -313,8 +313,7 @@ static ssize_t sof_dfsentry_trace_read(struct file *file, char __user *buffer,
 	lpos_64 = lpos;
 	lpos = do_div(lpos_64, buffer_size);
 
-	if (count > buffer_size - lpos) /* min() not used to avoid sparse warnings */
-		count = buffer_size - lpos;
+	count = min_t(size_t, count, buffer_size - lpos);
```

Comment:
when I see code like `a = min(a, b)` I always have to think that it unwinds to
```
if (a > b)
	a = b;
else
	a = a;
```
i.e. it has that superfluous assignment in it, so I often prefer just an explicit
```
if (a > b)
	a = b;
```
even though I realise, that the compiler probably optimises away the redundancy...

Reply:
Right, I'll drop the comment then from here and use the same `if (a > b) a = b;` fort he second change.

Reply:
After a second look at the code: I have dropped the first adjustment for the count as the second one is enough. The count is not used in-between them.